### DISCUSSION TITLE
Data migration flow

### DISCUSSION THREAD

**Discussion by michaltaberski** [pocketbase/pocketbase#1604] [Q&A] [answered]
Hi, 

I can see and understand the generated migration files related to data structure update - That's cool.

I wonder how could I make a flow to iterate over the existing data and perform update (unless it was already executed in the previous run).

What type of action might I want to run?
- after new field (let's say required*) has been added I want to run a script to iterate over existing rows and add a default value for them - just so I have valid consistent data where all required fields are filled up, so are really valid.

**Comment by ganigeorgiev** [comment, marked as answer]
You can create custom migrations in addition to the autogenerated ones. They are executed in the order of the unapplied migration filenames (_each migration file is prefixed by default with a timestamp_).

For details how to write migrations in Go you can check https://pocketbase.io/docs/migrations/#migrate-create.

For JS currently this is not documented (_it is left for v0.13/v0.14 together with the planned JS app hooks_), but the JS API mirrors the Go one with 2 caveats:

- the exported methods are camelCased
- errors are thrown and not returned as value

Here is a JS example flow:

1. Run `./pocketbase migrate create 'set_default_example_status'`
2. Edit the previously created blank migration file:
  ```js
  migrate((db) => {
    // set a default "pending" status to all example records
    db.newQuery("UPDATE example SET status = 'pending' WHERE status = ''").execute()
  }, (db) => {
    // no reverse...
  })
  ```
